authors/apps/authentication/views.py

```diff
@@ -24,10 +31,43 @@ def post(self, request):
         # your own work later on. Get familiar with it.
         serializer = self.serializer_class(data=user)
         serializer.is_valid(raise_exception=True)
-        serializer.save()
-
+        serializer.save()        
+
+        # Email verification email configuration via
+        # gmail API
+        email = serializer.data.get('email', None)
+        payload = {'email':  email}
+        token = jwt.encode(payload,SECRET_KEY, algorithm='HS256').decode()
+        from_email, to_email =EMAIL_HOST_USER , [email]
+        subject = "Authors Haven Verification Link"
+        site_url = get_current_site(request).domain
+        link_url = "http://" + site_url + \
+            '/api/users/verify/{}/'.format(token)
+        username = serializer.data['username']
+
+        message = render_to_string(
+            'email-verify.html', {
+                'user': to_email,
+                'domain': link_url,
+                'token': token,
+                'username': username,
+                'link': link_url
+            })
+
+        send_mail(subject,
+                'Please click on this link to verify your email.',from_email,
+                    [to_email, ], html_message=message, fail_silently=False)
         return Response(serializer.data, status=status.HTTP_201_CREATED)
 
+class VerifyAPIView(generics.ListCreateAPIView):
+    def get_queryset(self):
+        token = jwt.decode(
+            self.kwargs["token"],SECRET_KEY, algorithm='HS256')
+        queryset = User.objects.filter(email=token['email'])
+        User.objects.filter(email=token['email']).update(is_verified=True)
```

Comment:
since we are expecting only one User, why don't we use `.get`

Reply:
Done